app/src/editor/dashboard/components/DashboardModuleSearch.jsx

```diff
@@ -76,7 +76,11 @@ class DashboardModuleSearch extends React.PureComponent {
     async load() {
         const canvases = await getCanvases()
         if (this.unmounted) { return }
-        this.setState({ canvases })
+        this.setState({ canvases }, () => {
+            if (this.input) {
+                this.input.focus()
+            }
+        })
```

Comment:
I feel like the input focus should occur immediately upon opening the modal, rather than being tied to a potentially slow api to response? i.e. Allow user to at least start typing before the list of canvases is actually loaded.

Try this out with a latent e.g. 3s connection, it feels pretty bad.

Reply:
Yea, not ideal. I put it here because the focusing didn't work otherwise. It focused only once the rendered contents were actually on the screen.

Reply:
Hmm, does the focusing fail because the input ref is not set?

Reply:
I've added a short timeout to the focus which is hacky but the problem seems to be that due to the modal loading logic, the component is not instantly visible and React doesn't allow to focus on elements which are not visible.